client.go

```diff
@@ -50,26 +77,30 @@ type userResponse struct {
 	UserId   string `json:"Id"`
 }
 
-func (c *Client) Login(username, password string) error {
-	body := map[string]string{}
-	body["Username"] = username
-	body["PW"] = password
+// Login authenticates a user into the server provided in Client.
+// If the login is successful, the access token is stored for future API calls.
+func (c *Client) Login(ctx context.Context, username, password string) error {
```

Comment:
Adding the context here is a public API change. I am totally good with adding contexts - and it's definitely the "right" API to expose - but I think it should be done all at once rather than piecemeal (so I only have to refactor once in Supersonic :) So for now I'd prefer to hold off on adding contexts.

Reply:
No problem! Will remove.

Reply:
updated! Also change the url path parsing to be the same as `makeDo`